PathKit/Sources/LineSegment.swift

```diff
@@ -27,13 +27,64 @@ public struct LineSegment {
         return lerp(start, end, at: t)
     }
 
-    /// Creates a line segment begins at the point `start` and ends the point `end`.
+    /// Creates a line segment beginning at `start` and ending at `end`.
     ///
     /// - Parameters:
     ///   - start: The starting point of the line segment.
     ///   - end: The end point of the line segment.
+    /// - Precondition: The `start` point and `end` point of the line segment must be different.
     public init(start: Point, end: Point) {
+        precondition(start != end)
         self.start = start
-        self.end = start
+        self.end = end
+    }
+
+
+    public func offset(by vector: Vector) -> LineSegment {
+        return LineSegment(start: start + vector, end: end + vector)
+    }
+
+    /// Computes the intersection of two line segments.
+    /// Source: https://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
+    public static func intersection(_ lhs: LineSegment, _ rhs: LineSegment, accuracy: Double) -> Intersection {
+        let p = lhs.start
+        let q = rhs.start
+        let qp = Vector(head: q, tail: p)
```

Comment:
Let's come up with a better name for this. Something like `headsVector`

Reply:
ok will do